iaso/api/xls_form_generator.py

```diff
@@ -0,0 +1,240 @@
+    of the xlsform. It's tagged as 'insert_field' in the xlsform.
+    """
+    authorized_fields = [
+        "id",
+        "epid",
+        "obr_name",
+        "gpei_email",
+        "description",
+        "creation_email_send_at",
+        "onset_at",
+        "three_level_call_at",
+        "cvdpv_notified_at",
+        "cvdpv2_notified_at",
+        "pv_notified_at",
+        "pv2_notified_at",
+        "virus",
+        "detection_status",
+        "detection_responsible",
+        "detection_first_draft_submitted_at",
+        "detection_rrt_oprtt_approval_at",
+        "risk_assessment_status",
+        "risk_assessment_responsible",
+        "investigation_at",
+        "risk_assessment_first_draft_submitted_at",
+        "risk_assessment_rrt_oprtt_approval_at",
+        "ag_nopv_group_met_at",
+        "dg_authorized_at",
+        "verification_score",
+        "doses_requested",
+        "preperadness_spreadsheet_url",
+        "preperadness_sync_status",
+        "surge_spreadsheet_url",
+        "country_name_in_surge_spreadsheet",
+        "budget_status",
+        "budget_responsible",
+        "created_at",
+        "updated_at",
+        "district_count",
+        "round_one",
+        "round_two",
+        "vacine",
+        "obr_name",
+    ]
+    campaign = get_object_or_404(Campaign, id=campaign_id)
+    for i in range(2, row + 1):
+        cell_obj = q_sheet.cell(row=i, column=2)
+        cell_value_start = cell_obj.value[:7] if cell_obj.value is not None else ""
+        if cell_value_start == "insert_":
+            str_request = cell_obj.value[7:]
+            if str_request in authorized_fields:
+                cell_obj = q_sheet.cell(row=i, column=calculation_index)
+                cell_obj.value = str(getattr(campaign, str_request))
+
+
+def get_column_position(column_name, sheet):
+    for row_calc in sheet.rows:
+        column_position = 0
+        for s_cell in row_calc:
+            column_position += 1
+            if str(s_cell.value).lower() == column_name:
+                return column_position
+
+
+def create_ou_tree_list(group_ou):
+    # create list of dictionary with OU tree
+    ou_tree_list = []
+    for ou in group_ou:
+        ou_tree_dict = {ou.org_unit_type.name: ou for ou in ou.create_ou_tree()}
+        ou_tree_list.append(ou_tree_dict)
+    return ou_tree_list
+
+
+class XlsFormGeneratorViewSet(ModelViewSet):
+    results_key = "xlsformgenerator"
+    remove_results_key_if_paginated = True
+    filter_backends = [filters.OrderingFilter, DjangoFilterBackend, DeletionFilterBackend]
+
+    def get_serializer_class(self):
+        return XlsFormTemplateSerializer
+
+    def get_queryset(self):
+        queryset = XlsFormTemplate.objects.filter(account=self.request.user.iaso_profile.account)
+        return queryset
+
+    @action(methods=["GET"], detail=False)
+    def xlsform_generator(self, request) -> HttpResponse:
```

Comment:
This method is really long and focus on many things (building the request, creating a XLS file, moving data to it, ...). Would it be possible to refactor so it's shorter and more readable?